File: Level 2/조이스틱/solution.js
function solution(name) {
  let totalCountChangeChar = 0;
  let minMoveCount = name.length - 1;

  [...name].forEach((char, index) => {
    totalCountChangeChar += countChangeChar(char);

    let nextIndex = index + 1;

    while (nextIndex < name.length && name[nextIndex] === "A") {
      nextIndex++;
    }

    minMoveCount = Math.min(
      minMoveCount,
      index * 2 + name.length - nextIndex,
      index + 2 * (name.length - nextIndex)
    );
  });

  return totalCountChangeChar + minMoveCount;
}

function countChangeChar(char) {
  const moveTopCount = char.charCodeAt() - "A".charCodeAt();
  const moveBottomCount = "Z".charCodeAt() + 1 - char.charCodeAt();
  return Math.min(moveTopCount, moveBottomCount);
}

File: Level 2/조이스틱/solution.py
def solution(name):
    answer = 0
    min_move = len(name) - 1

    for index, char in enumerate(name):
        answer += min(ord(char) - ord("A"), ord("Z") + 1 - ord(char))

        next_index = index + 1
        while next_index < len(name) and name[next_index] == "A":
            next_index += 1

        min_move = min(
            min_move,
            index * 2 + (len(name) - next_index),
            (len(name) - next_index) * 2 + index,
        )

    return answer + min_move
